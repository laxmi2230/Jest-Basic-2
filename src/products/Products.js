import React from 'react'

const Products = ({ items = [] }) => {
    if (items.length <= 0) {
        return (
            <div>
                No items found
            </div>
        )
    } else {
        return <div>
            {items.map((item, index) => {
                return <div className="products" key={index}><h1>{item}</h1></div>
            })}
        </div>
    }
}

export default Products
