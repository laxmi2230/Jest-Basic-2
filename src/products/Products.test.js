import render from 'dom-serializer';
import React from 'react';
import renderer from 'react-test-renderer';

import Products from './Products';

it("renders when there are no items", () => {
    const tree = renderer.create(<Products />).toJSON();
    expect(tree).toMatchSnapshot();
})

it("renders when there is one item", () => {
    const tree = renderer.create(<Products items={["Mango"]} />).toJSON();
    expect(tree).toMatchSnapshot();
})

it("renders when there are multiple items", () => {
    const tree = renderer.create(<Products items={['Earth', 'mango', 'Moon', 'Saturn']} />).toJSON();
    expect(tree).toMatchSnapshot();
})