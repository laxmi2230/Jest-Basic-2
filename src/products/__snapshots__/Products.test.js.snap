// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when there are multiple items 1`] = `
<div>
  <div
    className="products"
  >
    <h1>
      Earth
    </h1>
  </div>
  <div
    className="products"
  >
    <h1>
      mango
    </h1>
  </div>
  <div
    className="products"
  >
    <h1>
      Moon
    </h1>
  </div>
  <div
    className="products"
  >
    <h1>
      Saturn
    </h1>
  </div>
</div>
`;

exports[`renders when there are no items 1`] = `
<div>
  No items found
</div>
`;

exports[`renders when there is one item 1`] = `
<div>
  <div
    className="products"
  >
    <h1>
      Mango
    </h1>
  </div>
</div>
`;
